scalar/list.go

```diff
@@ -93,7 +93,11 @@ func (s *List) Set(val any) error {
 		s.Value = make(Vector, length)
 		for i := 0; i < length; i++ {
 			s.Value[i] = NewScalar(s.Type.(*arrow.ListType).Elem())
-			if err := s.Value[i].Set(reflectedValue.Index(i).Interface()); err != nil {
+			iVal := reflectedValue.Index(i)
+			if (iVal.Kind() == reflect.Ptr || iVal.Kind() == reflect.Map || iVal.Kind() == reflect.Slice) && iVal.IsNil() {
+				continue
+			}
```

Comment:
```suggestion
			switch iVal.Kind() {
			case reflect.Pointer, reflect.UnsafePointer, reflect.Map, reflect.Slice, reflect.Interface:
				if iVal.IsNil() {
					continue
				}
			}
```

or, better yet, define a helper
```go
func isReflectValueNil(v reflect.Value) bool {
	switch iVal.Kind() {
	case reflect.Pointer,
		reflect.UnsafePointer,
		reflect.Map,
		reflect.Slice,
		reflect.Interface:
	default:
		return false
	}
	return v.IsNil()
}
```